### ISSUE TITLE
Fully support commonmark

### ISSUE THREAD

**Issue by leipert** [wooorm/mdast-lint#20]
If I set `commonmark: true` in mdast, does it affect the linter? So that some rules are turned off/on?

Furthermore I recognized, that nested blockquotes are not linted correctly:

<img width="787" alt="screen shot 2015-08-04 at 10 47 25" src="https://cloud.githubusercontent.com/assets/2906107/9056881/48de30fa-3a96-11e5-8285-c9daa99af5d9.png">

**Comment by leipert** [comment]
Okay, that blockquote error is gone in 0.4.3 :+1:

**Comment by wooorm** [comment]
Re your last example, you can fix that by turning it into:

```markdown
# Test

> Test
>
>> test again.
```

Which IMHO is actually better, and the warning was correct.

**Comment by leipert** [comment]
Okay. Good to know :)

**Comment by wooorm** [comment]
Regarding CommonMark: Yes, Commonmark is detected:

If I use this:

test.md:
```markdown
A paragraph.
A heading in GFM.
=================
```

.mdastrc:
```json
{
  "settings": {
    "commonmark": true
  }
}
```

...I get no warnings (because the heading and its rule is part of the paragraph).

However, without CommonMark:

test.md:
```markdown
A paragraph.
A heading in GFM.
=================
```

.mdastrc:
```json
{}
```

I get this (because it’s seen as a heading):
![screen shot 2015-08-04 at 11 03 24 a m](https://cloud.githubusercontent.com/assets/944406/9057183/74ddc8d0-3a98-11e5-8dc3-3fbf51b4bd6c.png)

Regarding fully supporting CommonMark, more info can be seen in [`mdast-html#commonmark`](https://github.com/wooorm/mdast-html#commonmark). It will take a lot of work, and I’ve got enough on my plate already so it’ll take some more time. Sorry!
